docs/02-app/02-api-reference/05-next-config-js/ppr.mdx

```diff
@@ -1,39 +1,87 @@
+## Using Partial Prerendering
 
-To enable Partial Prerendering, set the experimental `ppr` flag to `incremental`:
+### Incremental Adoption (Version 15)
 
-```js filename="next.config.js"
-/** @type {import('next').NextConfig} */
-const nextConfig = {
+In Next.js 15, you can incrementally adopt Partial Prerendering in [layouts](/docs/app/building-your-application/routing/layouts-and-templates) and [pages](/docs/app/building-your-application/routing/pages) by setting the [`ppr`](/docs/app/api-reference/next-config-js/ppr) option in `next.config.js` to `incremental`, and exporting the `experimental_ppr` [route config option](/docs/app/api-reference/file-conventions/route-segment-config) at the top of the file:
+
+```js filename="next.config.ts"
+import type { NextConfig } from 'next'
+
+const nextConfig: NextConfig = {
   experimental: {
-    ppr: 'incremental',
-  },
+    ppr: 'incremental'
+  }
 }
 
-module.exports = nextConfig
+export default nextConfig
 ```
 
-## Options
+```tsx filename="app/page.tsx" switcher
+import { Suspense } from "react"
+import { StaticComponent, DynamicComponent, Fallback } from "@/app/ui"
 
-- `incremental`: Allows you to incrementally opt-in layouts and pages, and should be used with the [`experimental_ppr`](/docs/app/api-reference/file-conventions/route-segment-config#experimental_ppr) route config option.
-- `true`: Enables Partial Prerendering for all routes in your application.
+export const experimental_ppr = true
 
-We recommend using `incremental` first. Once all the routes in your application have PPR enabled, it’ll be considered safe for you to set the `ppr` value to `true`, and enable it for the entire app and all future routes.
+export default function Page() {
+  return {
+     <>
+      <StaticComponent />
+      <Suspense fallback={<Fallback />}>
+        <DynamicComponent />
+      </Suspense>
+     </>
+  };
+}
+```
+
+```jsx filename="app/page.js" switcher
+import { Suspense } from "react"
+import { StaticComponent, DynamicComponent, Fallback } from "@/app/ui"
 
-> **Good to know:**
+export const experimental_ppr = true
+
+export default function Page() {
+  return {
+     <>
+      <StaticComponent />
+      <Suspense fallback={<Fallback />}>
+        <DynamicComponent />
+      </Suspense>
+     </>
+  };
+}
+```
+
+> **Good to know**:
 >
-> - When using `incremental`, routes that don't have `experimental_ppr` will default to `false` and will not be prerendered using Partial Prerendering. You need to explicitly opt-in to Partial Prerendering for each route.
-> - Partial Prerendering does not yet apply to client-side navigations. We are actively working on this.
-> - Partial Prerendering is designed for the [Node.js runtime](/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes) only. Using the Edge Runtime is not needed when you can instantly serve the static shell.
```

Comment:
@leerob I think it is helpful to clarify in this documentation that PPR currently works in Node so it is clear before a user hits [this warning](https://github.com/vercel/next.js/blob/121bd6a199b5b5a718f0846691cc31324aab9918/packages/next/src/build/index.ts#L2124). It would also be great to document here the future plans for PPR for supporting the edge runtime. [There is a TODO in the code](https://github.com/vercel/next.js/blob/121bd6a199b5b5a718f0846691cc31324aab9918/packages/next/src/build/index.ts#L2118) to give support for the edge runtime, and if that is indeed in the works then it'd be great to call it out here! 

I'm very much in favor of Next supporting the edge runtime for PPR and vouch for keeping it on the near-term roadmap. A lot of things make sense to run on origin servers / in Node, but PPR at the edge is extremely powerful.

Reply:
Thanks Max. PPR is still experimental and we're focused on the Node.js runtime based on customer demand. It's possible that demand might change in the future and we can reassess. I'd still like to remove this note because it's accidentally conflating the _runtime_ and the _location_, which is something historically we have not been precise enough about.